.github/workflows/ci.yml

```diff
@@ -147,3 +147,19 @@ jobs:
         run: |
           git-secrets --register-aws
           git-secrets --scan
+  custom-header:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone This Repo
+        uses: actions/checkout@v2
+      - name: Build
+        run: |
+          mkdir -p override-include
+          cp source/include/stdbool.readme override-include/stdbool.h
+          cp source/include/stdint.readme override-include/stdint.h
+          cmake -S test -B build/ \
+          -G "Unix Makefiles" \
+          -DCMAKE_BUILD_TYPE=Debug \
```

Comment:
This is conflicting with the `-DNDEBUG` flag passed. Should that flag be passed?

Reply:
I just copied the steps of the `unittest` job, we do this so that asserts don't show up in coverage. I could remove one or both and it shouldn't really affect anything since we're only testing if it builds without errors

Reply:
Maybe the `unittest` job should mention the build type as `RELEASE` which (I think) would automatically take care of providing the `-DNDEBUG` flag to the compiler.
That is beyond the scope of this PR though, so will not block on it.